ledger/voters.go

```diff
@@ -156,36 +154,48 @@ func (vt *votersTracker) close() {
 	vt.loadWaitGroup.Wait()
 }
 
-func (vt *votersTracker) newBlock(hdr bookkeeping.BlockHeader) {
-	proto := config.Consensus[hdr.CurrentProtocol]
+func (vt *votersTracker) newBlock(blk bookkeeping.Block, _ ledgercore.StateDelta) {
+	proto := config.Consensus[blk.CurrentProtocol]
 	if proto.CompactCertRounds == 0 {
 		// No compact certs.
 		return
 	}
 
+	vt.hdr = blk.BlockHeader
+
 	// Check if any blocks can be forgotten because the compact cert is available.
 	for r, tr := range vt.round {
 		commitRound := r + basics.Round(tr.Proto.CompactCertVotersLookback)
 		certRound := commitRound + basics.Round(tr.Proto.CompactCertRounds)
-		if certRound < hdr.CompactCert[protocol.CompactCertBasic].CompactCertNextRound {
+		if certRound < vt.hdr.CompactCert[protocol.CompactCertBasic].CompactCertNextRound {
 			delete(vt.round, r)
 		}
 	}
 
 	// This might be a block where we snapshot the online participants,
 	// to eventually construct a merkle tree for commitment in a later
 	// block.
-	r := uint64(hdr.Round)
+	r := uint64(vt.hdr.Round)
 	if (r+proto.CompactCertVotersLookback)%proto.CompactCertRounds == 0 {
 		_, ok := vt.round[basics.Round(r)]
 		if ok {
 			vt.l.trackerLog().Errorf("votersTracker.newBlock: round %d already present", r)
 		} else {
-			vt.loadTree(hdr)
+			vt.loadTree(vt.hdr)
 		}
 	}
 }
 
+func (vt *votersTracker) committedUpTo(committedRound basics.Round) (minRound, lookback basics.Round) {
+	proto := config.Consensus[vt.hdr.CurrentProtocol]
```

Comment:
Why not just fetch this from vt.l.BlockHdr(committedRound)?  Otherwise I don't understand why this is correct: what round are you expecting vt.hdr to be at?  What if vt.hdr is much further along (or much further behind)?